scapy/fields.py

```diff
@@ -1804,36 +1807,27 @@ def randval(self):
         return RandBin(RandNum(0, self.max_length or 1200))
 
 
-class XStrField(StrField):
-    """
-    StrField which value is printed as hexadecimal.
-    """
-
+class _XStrField:
     def i2repr(self, pkt, x):
         # type: (Optional[Packet], bytes) -> str
         if isinstance(x, bytes):
             return bytes_hex(x).decode()
-        return super(XStrField, self).i2repr(pkt, x)
+        return super(_XStrField, self).i2repr(pkt, x)
```

Comment:
I don't get this one: `_XStrField` only inherits from `object` which lacks `.i2repr()` method. Or did I miss something?
Maybe:
```suggestion
        return Field.i2repr(self, pkt, x)
```

Reply:
You're right I forgot something